src/client.js

```diff
@@ -37,59 +40,42 @@ function podmanMonitor(name, method, args, callback, system) {
 }
 
 export function streamEvents(system, callback) {
-    return new Promise((resolve, reject) => {
-        podmanMonitor("libpod/events", "GET", {}, callback, system)
-                .then(reply => resolve(JSON.parse(reply)))
-                .catch(reject);
-    });
+    return podmanMonitor("libpod/events", "GET", {}, callback, system)
+            .then(reply => JSON.parse(reply));
 }
 
 export function getInfo(system) {
     return new Promise((resolve, reject) => {
         const timeout = setTimeout(() => reject(new Error("timeout")), 5000);
-        podmanCall("libpod/info", "GET", {}, system)
-                .then(reply => resolve(JSON.parse(reply)))
+        podmanJson("libpod/info", "GET", {}, system)
+                .then(reply => resolve(reply))
                 .catch(reject)
                 .finally(() => clearTimeout(timeout));
     });
 }
 
 export function getContainers(system, id) {
-    return new Promise((resolve, reject) => {
-        const options = { all: true };
-        if (id)
-            options.filters = JSON.stringify({ id: [id] });
+    const options = { all: true };
+    if (id)
+        options.filters = JSON.stringify({ id: [id] });
 
-        podmanCall("libpod/containers/json", "GET", options, system)
-                .then(reply => resolve(JSON.parse(reply)))
-                .catch(reject);
-    });
+    return podmanJson("libpod/containers/json", "GET", options, system);
 }
 
 export const getContainerStats = (system, callback) => podmanMonitor("libpod/containers/stats", "GET", { stream: true }, callback, system);
 
 export function inspectContainer(system, id) {
-    return new Promise((resolve, reject) => {
-        const options = {
-            size: false // set true to display filesystem usage
-        };
-        podmanCall("libpod/containers/" + id + "/json", "GET", options, system)
-                .then(reply => resolve(JSON.parse(reply)))
-                .catch(reject);
-    });
+    const options = {
+        size: false // set true to display filesystem usage
+    };
+    return podmanJson("libpod/containers/" + id + "/json", "GET", options, system);
 }
 
 export const delContainer = (system, id, force) => podmanCall("libpod/containers/" + id, "DELETE", { force }, system);
 
 export const renameContainer = (system, id, config) => podmanCall("libpod/containers/" + id + "/rename", "POST", config, system);
 
-export function createContainer(system, config) {
-    return new Promise((resolve, reject) => {
-        podmanCall("libpod/containers/create", "POST", {}, system, JSON.stringify(config))
-                .then(reply => resolve(JSON.parse(reply)))
-                .catch(reject);
-    });
-}
+export const createContainer = (system, config) => podmanJson("libpod/containers/create", "POST", {}, system, JSON.stringify(config));
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1339-20230706-033702-672c013e-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)